flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/parser/CustomAlterTableParserListener.java

```diff
@@ -315,7 +319,7 @@ public void exitAlterByChangeColumn(MySqlParser.AlterByChangeColumnContext ctx)
                     String newColumnName = parser.parseName(ctx.newColumn);
 
                     Map<String, DataType> typeMapping = new HashMap<>();
-                    typeMapping.put(column.name(), fromDbzColumn(column));
+                    typeMapping.put(column.name(), fromDbzColumn(column, jdbcProperties));
```

Comment:
I think we can not convert the tinyint(1) to boolean when it is from the binlog.

A added tinyint(1) column can be parsed right from the binlog. If it is changed here, the sink is impossible to add a tinyint(1) column.

A added boolean column can be parsed right as boolean from the binlog. Please test it and add some tests to cover this alter sql.